libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -369,6 +369,29 @@ void HTTPClient::setTimeout(uint16_t timeout)
     }
 }
 
+/**
+ * set the URL to a new value. Handy for following redirects.
+ * @param url
+ */
+bool HTTPClient::setURL(String url)
+{
+    // check for : (http: or https:)
+    int index = url.indexOf(':');
+    if(index < 0) {
+        DEBUG_HTTPCLIENT("[HTTP-Client][begin] failed to parse protocol\n");
+        return false;
+    }
+
+    String protocol = url.substring(0, index);
+    if(protocol != "http" && protocol != "https") {
+        DEBUG_HTTPCLIENT("[HTTP-Client][begin] unknown protocol '%s'\n", protocol.c_str());
+        return false;
+    }
+
+    _port = (protocol == "https" ? 443 : 80);
+    return beginInternal(url, protocol.c_str());
+}
+
```

Comment:
Is there any way to factorize parsing code used in `begin(url...)` ?

Reply:
I'll take a look this weekend.

Reply:
So i've refactored a bit. It does bring up the question of when the redirect is from 'http' to 'https', or vise versa, as traits are incorrect in the current version and when #4980 is merged the current client would not support the new protocol.  So, i'm wondering if #4980 should support a ClientFactory parameter so that the client could be created/recreated when the protocol changes.  Or maybe its not worth the work and we just have to make a new HTTPClient for redirects that change protocol. 
Currently I have setURL() returning false if the protocol would change.